atc/atccmd/command.go

```diff
@@ -184,6 +184,7 @@ type RunCommand struct {
 		OneOffBuildGracePeriod time.Duration `long:"one-off-grace-period" default:"5m" description:"Period after which one-off build containers will be garbage-collected."`
 		MissingGracePeriod     time.Duration `long:"missing-grace-period" default:"5m" description:"Period after which to reap containers and volumes that were created but went missing from the worker."`
 		HijackGracePeriod      time.Duration `long:"hijack-grace-period" default:"10m" description:"Period after which hijacked containers will be garbage collected"`
+		FailedGracePeriod      time.Duration `long:"failed-grace-period" default:"4h" description:"Period after which failed containers will be garbage collected"`
```

Comment:
@vito what is a better default value here that makes less impact to the GC?

Also @ebilling could you add a check for this flag so when it is set to '0' the failed containers of last build will still be kept as of today (i.e. user opt out this period) ? There might be users who got sufficient resource that dont mind keeping the failed containers longer. Thx!

Reply:
@xtremerui, made the change to go back to current behavior if failed-grace-period set to '0'.

Reply:
Hmm, I think 4 hours is too short at least. One of the reasons for keeping them forever was so that a build that failed on a Friday evening could still be debugged Monday morning. Build failures can be lightning in a bottle, so allowing the build to be debugged for longer periods seemed important at the time.

"Infinite" may be too costly for how infrequently this situation occurs though, and never being able to reproduce a failure again may in itself be feedback that your pipeline is too brittle and needs more sweeping changes. That won't help the sting go away at the time though.

@ebilling Not necessarily suggesting it, but what would be the highest theoretical value you would allow here? `72h` would allow a flake to survive a weekend, while not being infinite. 🤔

Reply:
Let's make it 96h.  That gives almost a week or a long weekend to get in.

Reply:
Giving the case of keeping failed container to survive a weekend, I would say 24 hours after the weekend might still not be enough since Monday could be occupied by meetings plans etc. How about 120 hours (5 days) by default? So it will be available long enough for a work cycle of build failed->notified dev->planing fix->debugging->rerun build.

Reply:
Anything is better than infinite.  I'll push the change.

Reply:
just so it doesn't get forgotten (and in the interest of practicing responsible change), setting any default makes this a breaking change and it should be noted in the release notes!